rest_framework_gis/schema.py

```diff
@@ -0,0 +1,168 @@
+
+class GeoFeatureAutoSchema(AutoSchema):
+    COORDINATES_SCHEMA_FOR_POINT = {
+        'type': 'array',
+        'items': {
+            'type': 'number',
+            'format': 'float',
+        },
+        'example': [12.9721, 77.5933],
+        'minItems': 2,
+        'maxItems': 3,
+    }
+
+    COORDINATES_SCHEMA_FOR_LINE_STRING = {
+        'type': 'array',
+        'items': COORDINATES_SCHEMA_FOR_POINT,
+        'example': [[22.4707, 70.0577], [12.9721, 77.5933]],
+        'minItems': 2,
+    }
+
+    GEO_FIELD_TO_SCHEMA = {
+        models.PointField: {
+            'type': {
+                'type': 'string',
+                'enum': ['Point'],
+            },
+            'coordinates': COORDINATES_SCHEMA_FOR_POINT,
+        },
+        models.LineStringField: {
+            'type': {
+                'type': 'string',
+                'enum': ['LineString'],
+            },
+            'coordinates': COORDINATES_SCHEMA_FOR_LINE_STRING,
+        },
+        models.PolygonField: {
+            'type': {
+                'type': 'string',
+                'enum': ['Polygon'],
+            },
+            'coordinates': {
+                'type': 'array',
+                'items': {
+                    **COORDINATES_SCHEMA_FOR_LINE_STRING,
+                    'minItems': 4,
+                },
+                'example': [[0.0, 0.0], [0.0, 50.0], [50.0, 50.0], [50.0, 0.0], [0.0, 0.0]]
+            }
+        }
+    }
+
+    MULTI_FIELD_MAPPING = {
+        models.PointField: models.MultiPointField,
+        models.LineStringField: models.MultiLineStringField,
+        models.PolygonField: models.MultiPolygonField
+    }
+
+    for singular_field, multi_field in MULTI_FIELD_MAPPING.items():
+        GEO_FIELD_TO_SCHEMA[multi_field] = {
+            'type': {
+                'type': 'string',
+                'enum': [multi_field.geom_class.__name__],
+            },
+            'coordinates': {
+                'type': 'array',
+                'items': GEO_FIELD_TO_SCHEMA[singular_field]['coordinates'],
+                'example': [
+                    GEO_FIELD_TO_SCHEMA[singular_field]['coordinates']['example']
+                ]
+            }
+        }
+
+    def _map_geo_field(self, serializer, geo_field_name):
+        field = serializer.fields[geo_field_name]
+        if isinstance(field, GeometrySerializerMethodField):
+            warnings.warn('Geometry generation for GeometrySerializerMethodField is not supported.'.format(field=field))
+            return {}
+
+        model_field_name = geo_field_name
+
+        geo_field = model_meta.get_field_info(serializer.Meta.model).fields[model_field_name]
+        try:
+            return self.GEO_FIELD_TO_SCHEMA[geo_field.__class__]
+        except KeyError:
+            warnings.warn('Geometry generation for {field} is not supported.'.format(field=field))
+            return {}
+
+    def _map_field(self, field):
+        if isinstance(field, GeoFeatureModelListSerializer):
```

Comment:
this looks incorrect.  `_map_field` is a method from DRF, and the arg `field` is not a serializer.  I have to wonder how a serializer is appearing here.

Anyway, `_map_field` should handle `if isinstance(field, GeometryField):`.
I've added a commit https://github.com/jayvdb/django-rest-framework-gis/commit/85983bc0fbd7b811be2a95c53bf23223e8cabe9b with some of the things that seem needed for that.

I needed this when using spatialite, which may be a bit different under the cover than the postgres driver.

I have a travis build https://travis-ci.org/github/jayvdb/django-rest-framework-gis/jobs/682327850 and  `test_point_field_inner_geo_list_serializer` might be a real breakage caused by my changes.

Reply:
> this looks incorrect. `_map_field` is a method from DRF, and the arg `field` is not a serializer. I have to wonder how a serializer is appearing here.

Have a look at line no 364 of rest_framework/schemas/openapi.py

Reply:
> Anyway, `_map_field` should handle `if isinstance(field, GeometryField):`.

Line no 369 of `rest_framework/schemas/openapi.py` will handle this case automatically.